### PR TITLE
Add bonus difficulty if dataset is big

### PR DESCRIPTION
**Pull request by lhoestq**
Add a bonus difficulty of 20 for `duckdb-index` if the dataset is bigger than 3GB (from the config info).
This is defined in the orchestrator.

This can be used to defined "heavy workers" that would have enough RAM to process them (up to 29GB of memory for datasets like C4 that have their first 5GB converted to parquet)

The main changes are in
libs/libcommon/src/libcommon/config.py
libs/libcommon/src/libcommon/constants.py
libs/libcommon/src/libcommon/orchestrator.py
libs/libcommon/src/libcommon/processing_graph.py

TODO:
- [x] tests

Related to https://github.com/huggingface/datasets-server/issues/1891

### CODE REVIEW

File: libs/libcommon/src/libcommon/processing_graph.py [hunk 1]
```diff
@@ -593,8 +633,8 @@ def parse_id(id: str) -> tuple[str, str, Optional[str], Optional[str], str]:
         parts = parts[2:]
         config = None
         split = None
-        if len(parts) > 1:
-            config = parts[1]
-            if len(parts) > 2:
-                split = parts[2]
+        if len(parts) > 0:
```
Comment (severo) [inline libs/libcommon/src/libcommon/processing_graph.py:636]: it was wrong? and had no consequences until now?

File: libs/libcommon/src/libcommon/config.py [hunk 2]
```diff
@@ -35,7 +36,9 @@
     PROCESSING_STEP_SPLIT_OPT_IN_OUT_URLS_COUNT_VERSION,
     PROCESSING_STEP_SPLIT_OPT_IN_OUT_URLS_SCAN_VERSION,
 )
-from libcommon.processing_graph import ProcessingGraphSpecification
+
+if TYPE_CHECKING:
+    from libcommon.processing_graph import ProcessingGraphSpecification
```
Comment (severo) [inline libs/libcommon/src/libcommon/config.py:41]: why the change?

File: libs/libcommon/tests/test_orchestrator.py [hunk 3]
```diff
@@ -345,3 +346,80 @@ def test_remove_dataset() -> None:
     pending_jobs_df = Queue().get_pending_jobs_df(dataset=DATASET_NAME)
     assert len(pending_jobs_df) == 0
     assert has_some_cache(dataset=DATASET_NAME) is False
+
+
+@pytest.mark.parametrize("is_big", [False, True])
+def test_after_joplan_gives_bonus_difficulty(is_big: bool) -> None:
```
Comment (severo) [inline libs/libcommon/tests/test_orchestrator.py:352]: nit

```suggestion
def test_after_job_plan_gives_bonus_difficulty(is_big: bool) -> None:
```

**Review by severo — APPROVED** [review]

File: libs/libcommon/src/libcommon/config.py (hunk 2 above)
Comment (lhoestq) [reply libs/libcommon/src/libcommon/config.py:41]: To avoid circular dependencies

File: libs/libcommon/src/libcommon/processing_graph.py (hunk 1 above)
Comment (lhoestq) [reply libs/libcommon/src/libcommon/processing_graph.py:636]: It was wrong and had no consequences since this function was only used on jobs at dataset level in tests

Comment (severo) [reply libs/libcommon/src/libcommon/processing_graph.py:636]: OK, thanks for checking!

File: libs/libcommon/src/libcommon/config.py (hunk 2 above)
Comment (severo) [reply libs/libcommon/src/libcommon/config.py:41]: hmmm, ok. I imagine that's why I used the "specification" field at that time. Maybe it's better to move `ProcessingStepSpecification` and `ProcessingGraphSpecification` to a types file, no?

Comment (lhoestq) [reply libs/libcommon/src/libcommon/config.py:41]: I think it's fine.

It's common for config files to import other types using TYPE_CHECKING to make sure every file can still import the config

Comment (severo) [reply libs/libcommon/src/libcommon/config.py:41]: OKOK
